src/observe-tag.js

```diff
@@ -1,6 +1,24 @@
 const { observe } = require('@nx-js/observer-util')
 const { TRAM_TAG_REACTION, TRAM_TAG_NEW_EFFECTS, TRAM_TAG_CLEANUP_EFFECTS } = require('./node-names')
 
+// functions to go to nodes or indicies (made for .map)
+const toIndicies = (node, index) => index
+const toNodes = allNodes => index => allNodes[index]
+
+// sorting function that prioritizes indicies that are closest to a target
+// e.g. target = 3, [1, 2, 3, 4, 5] => [3, 2, 4, 1, 5]
+const byDistanceFromIndex = targetIndex => (indexA, indexB) => {
+	const diffFromTargetA = Math.abs(indexA - targetIndex)
+	const diffFromTargetB = Math.abs(indexB - targetIndex)
+	return diffFromTargetA < diffFromTargetB ? -1 : 1
+}
+
+// get an array including the element and all it's children
+const parentAndChildrenElements = node => {
+	const children = node.querySelectorAll('*')
+	return [node, ...children]
+}
```

Comment:
Is this recursive/ should it be?

Reply:
`querySelectorAll` (which we use in other places as well) will actually get all the children under an element.
![image](https://user-images.githubusercontent.com/326557/126072479-da4d58d4-31a8-44b5-a015-39f6b11688d1.png)